src/components/card/safe-title.tsx

```diff
@@ -0,0 +1,19 @@
+const setSafeHeadingLevel = (h: string): any => {
+  const validHeadingLevels = ['h1', 'h2', 'h3', 'h4']
+  const safeHeading = h ? h.toLowerCase() : ''
+
+  return validHeadingLevels.includes(safeHeading) ? safeHeading : 'p'
+}
+const SafeTitle: React.FC<{headingLevel: string; className?: string}> = ({
+  children,
+  headingLevel,
+  className = '',
+}) => {
+  const Heading = setSafeHeadingLevel(headingLevel) as React.FC<{
+    children: React.ReactNode
+    className?: string
+  }>
+  return <Heading className={className}>{children}</Heading>
+}
+
+export default SafeTitle
```

Comment:
The best solution I found for dynamically setting a heading. TypeScript does not like the conversion from string to ReactNode.

Reply:
I also restricted heading options to 4 levels because that seems deep enough in 99% of cases. Want to add some friction to more heading levels